feature-libs/quote/components/actions/by-role/quote-actions-by-role.component.ts

```diff
@@ -0,0 +1,235 @@
+      ?.pipe(take(1))
+      .subscribe();
+  }
+
+  protected handleConfirmationDialogClose(
+    action: QuoteActionType,
+    context: ConfirmationContext
+  ) {
+    this.subscription.unsubscribe();
+    this.subscription = new Subscription();
+    this.subscription.add(
+      this.launchDialogService.dialogClose
+        .pipe(
+          filter((reason) => reason === 'yes'),
+          tap(() => this.performAction(action, context.quote)),
+          filter(() => !!context.successMessage),
+          tap(() =>
+            this.globalMessageService.add(
+              { key: context.successMessage },
+              this.getMessageType(action)
+            )
+          )
+        )
+        .subscribe()
+    );
+  }
+
+  protected getMessageType(action: QuoteActionType): GlobalMessageType {
+    return action === QuoteActionType.CANCEL ||
+      action === QuoteActionType.REJECT
+      ? GlobalMessageType.MSG_TYPE_INFO
+      : GlobalMessageType.MSG_TYPE_CONFIRMATION;
+  }
+
+  protected requote(quoteId: string) {
+    this.quoteFacade.requote(quoteId);
+  }
+
+  getButtonStyle(allowedActions: QuoteAction[], action: QuoteAction): string {
+    if (action.isPrimary) {
+      return 'btn-primary';
+    }
+    if (allowedActions.length <= 2) {
+      return 'btn-secondary';
+    }
+    return action.type === QuoteActionType.CANCEL
+      ? 'btn-tertiary'
+      : 'btn-secondary';
+  }
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+
+  protected isConfirmationDialogRequired(
+    action: QuoteActionType,
+    state: QuoteState,
+    cartIsEmpty: boolean
+  ): boolean {
+    const mappingConfig = this.quoteUIConfig.quote?.confirmActionDialogMapping;
+    const dialogConfig =
+      mappingConfig?.[state]?.[action] ??
+      mappingConfig?.[this.stateToRoleTypeForDialogConfig(state)]?.[action] ??
+      mappingConfig?.[QuoteRoleType.ALL]?.[action];
+    return (
+      !!dialogConfig &&
+      (!cartIsEmpty || !dialogConfig.showOnlyWhenCartIsNotEmpty)
+    );
+  }
+
+  protected prepareConfirmationContext(
+    action: QuoteActionType,
+    quote: Quote
+  ): ConfirmationContext {
+    const dialogConfig = this.getDialogConfig(action, quote.state);
+    const confirmationContext: ConfirmationContext = {
+      quote: quote,
+      title: dialogConfig.i18nKeyPrefix + '.title',
+      confirmNote: dialogConfig.i18nKeyPrefix + '.confirmNote',
+      a11y: {
+        close: dialogConfig.i18nKeyPrefix + '.a11y.close',
+      },
+    };
+    if (dialogConfig.showWarningNote) {
+      confirmationContext.warningNote =
+        dialogConfig.i18nKeyPrefix + '.warningNote';
+    }
+    if (dialogConfig.showExpirationDate) {
+      confirmationContext.validity = 'quote.actions.confirmDialog.validity';
+    }
+    if (dialogConfig.showSuccessMessage) {
+      confirmationContext.successMessage =
+        dialogConfig.i18nKeyPrefix + '.successMessage';
+    }
+    return confirmationContext;
+  }
+
+  protected getDialogConfig(
```

Comment:
Please, either rename this method to `getConfirmationDialogConfig` or remove `Confirmation` word from other methods

Reply:
@Uli-Tiger, could you have a look?